### ISSUE TITLE
16~00 compiles to bound function instead of 0

### ISSUE THREAD

**Issue by ysulyma** [gkz/LiveScript#509]

**Comment by pepkin88** [comment]
It is this line https://github.com/gkz/LiveScript/blob/2c691670e065ac98a5964a3d51719cc92baa816b/src/lexer.ls#L304
Specifically `num is parse-int rnum.slice(0 -1), radix`
What was the rationale behind this check? After removing it, it passes all tests.

**Comment by rhendric** [comment]
I'd vote for 3, on the general principle of avoiding more special cases in the language (this would be the only language feature where `$` means something, other than its ubiquitous use as a behind-the-scenes name mangling character). But I don't generally use number suffixes.

**Comment by pepkin88** [comment]
The 1st option also allows using `$` sign without treating it as a special character. In that case this is only a convention preventing accidental comment leak (like in `16~08BAC_KSPACE`).
Although I guess this is not that important, because you can still use standard comments or constant names.

**Comment by rhendric** [comment]
Yeah, both 1 and 2 seem pretty objectionable to me—nobody should have to think about what characters are valid in base 27 in order to interpret the meaning of `27~stuff`. It should consistently mean ‘this is a number,’ which could possibly be invalid.
